codersdk/templates.go

```diff
@@ -513,3 +513,28 @@ func (c *Client) StarterTemplates(ctx context.Context) ([]TemplateExample, error
 	var templateExamples []TemplateExample
 	return templateExamples, json.NewDecoder(res.Body).Decode(&templateExamples)
 }
+
+type InvalidatePrebuildsResponse struct {
+	InvalidatedPresets []string `json:"invalidated_presets"`
+}
+
+// InvalidateTemplatePrebuilds invalidates all prebuilt workspaces for the
+// template's active version by setting last_invalidated_at timestamp.
+// The reconciler will then mark these prebuilds as expired and create new ones.
+func (c *Client) InvalidateTemplatePrebuilds(ctx context.Context, template uuid.UUID) (InvalidatePrebuildsResponse, error) {
```

Comment:
nit: Since we are actually invalidating the template's presets.
```suggestion
func (c *Client) InvalidateTemplatePresets(ctx context.Context, template uuid.UUID) (InvalidatePrebuildsResponse, error) {
```